Backend/app/rider-platform/rider-app/Main/src/SharedLogic/MerchantConfig.hs

```diff
@@ -114,11 +137,32 @@ checkFraudDetected riderId merchantId factors merchantConfigs = Redis.withCrossA
           cancelledBookingByDriverCount :: Int <- sum . catMaybes <$> SWC.getCurrentWindowValues (mkCancellationByDriverKey mc.id.getId riderId.getId) mc.fraudBookingCancelledByDriverCountWindow
           pure $ cancelledBookingByDriverCount >= mc.fraudBookingCancelledByDriverCountThreshold
         MoreSearching -> do
-          sreachCount :: Int <- sum . catMaybes <$> SWC.getCurrentWindowValues (mkSearchCounterKey mc.id.getId riderId.getId) mc.fraudSearchCountWindow
-          pure $ sreachCount >= mc.fraudSearchCountThreshold
+          searchCount :: Int <- sum . catMaybes <$> SWC.getCurrentWindowValues (mkSearchCounterKey mc.id.getId riderId.getId) mc.fraudSearchCountWindow
+          pure $ searchCount >= mc.fraudSearchCountThreshold
         TotalRides -> do
           totalRidesCount :: Int <- getTotalRidesCount riderId
           pure $ totalRidesCount <= mc.fraudBookingTotalCountThreshold
+        TotalRidesInWindow -> do
+          windowValueList <- SWC.getCurrentWindowValues (mkRideWindowCountKey mc.id.getId riderId.getId) mc.fraudRideCountWindow
+          let timeInterval = SWC.convertPeriodTypeToSeconds mc.fraudRideCountWindow.periodType
+          let intervals = enumFromThenTo 0 (fromIntegral timeInterval) ((length windowValueList -1) * fromIntegral timeInterval)
```

Comment:
you can use `[0, fromIntegral timeInterval .. ((length windowValueList -1) * fromIntegral timeInterval)]` syntax instead